malsim/mal_simulator.py

```diff
@@ -222,60 +222,137 @@ def reset(
 
         return self.agent_states
 
-    def _init_agent_action_surfaces(self):
-        """Set agent action surfaces according to current state"""
-        for agent in self._agent_states.values():
-
-            if isinstance(agent, MalSimAttackerState):
-                attacker = agent.attacker
-                agent.action_surface = query.calculate_attack_surface(
-                    attacker, skip_compromised = True
-                )
+    def _create_attacker_state(
+        self, name: str, attacker: Attacker
+    ) -> MalSimAttackerState:
```

Comment:
Splitting into create and update seems redundant. We could keep the latter only. When a new state is needed, we just create the object and pass that in the update method with the rest of the args being empty sets. The same for the defender.

If you do the above you may feel that creating manually a state object in register_attacker/register_defender and reset_agents does not look nice. The solution is for register_* to call reset_agents, until Andrei removes the Attackers completely. Then we can revisit if needed.

Reply:
I agree with what you both said and I'll re-evaluate it after the attacker logic has been moved and I see what it looks like. The reason I will leave it in for now is this blank state is needed in two places, when the simulator is created and upon reset.